### PR TITLE
feat: update how auth to hubspot api is done

### PR DESCRIPTION
**Pull request by matthugs**
Hubspot API needs us to pass its auth token as a header rather than as a query param.

credit from https://github.com/openedx/tubular/pull/650, getting this started

JIRA:PSRE-2394

### CODE REVIEW

**Review by ohnickmoy — APPROVED** [review]

File: tubular/hubspot_api.py [hunk 1]
```diff
@@ -58,10 +58,14 @@ def delete_user_by_vid(self, vid):
         """
         Delete a learner from hubspot using their Hubspot `vid` (unique identifier)
         """
+        headers = {
+            'content-type': 'application/json',
+            'authorization': 'Bearer %s' % {self.api_key}
```
Comment (adzuci) [inline tubular/hubspot_api.py]: We may have already sorted this out, but I believe this should be a new type of key which Taylor offered to help create for us with certain scoped permissions, is that correct and are there any remaining steps to get that new key?

**Review by adzuci — APPROVED** [review]
Is there a way to test this code locally with a key?  Possibly with help from marketing?

File: tubular/hubspot_api.py (hunk 1 above)
Comment (matthugs) [reply tubular/hubspot_api.py]: @adzuci Based on https://developers.hubspot.com/changelog/upcoming-api-key-sunset, yes it seems like that's correct.

I suspect whatever Taylor gives us would get plugged in here: https://github.com/edx-ops/user-retirement-secure/blob/c3319f3c9f54570634ecb5f6d75cdfce13819058/prod-edx.yml#L26
That is a code change we would likely need to synchronize with the merging of this PR.

**Comment by matthugs** [comment]
> Is there a way to test this code locally with a key? Possibly with help from marketing?

we _could_ do something like this: https://2u-internal.atlassian.net/wiki/spaces/DE/pages/10784609/How+To+Run+a+set+of+users+through+part+of+the+retirement+process

Since the pipeline step is idempotent we could test on a learner who was about to be deleted anyway and be unlikely to interrupt the pipeline running cleanly for that user if all goes well.

EDIT: so tested with a couple users; although it took some guess-and-check before I found a not-yet-deleted user who had any data in hubspot

File: tubular/hubspot_api.py (hunk 1 above)
Comment (matthugs) [reply tubular/hubspot_api.py]: addressed in https://github.com/edx-ops/user-retirement-secure/pull/219

**Pull request merged**